sdk/helper/aliasmetadata/alias_metadata.go

```diff
@@ -0,0 +1,195 @@
+	Default []string
+
+	// AvailableToAdd is a list of fields not included by
+	// default, that the user may include.
+	AvailableToAdd []string
+}
+
+func (f *Fields) all() []string {
+	return append(f.Default, f.AvailableToAdd...)
+}
+
+// FieldSchema takes the default and additionally available
+// fields, and uses them to generate a verbose description
+// regarding how to use the "alias_metadata" field.
+func FieldSchema(fields *Fields) *framework.FieldSchema {
+	return &framework.FieldSchema{
+		Type:        framework.TypeCommaStringSlice,
+		Description: description(fields),
+		DisplayAttrs: &framework.DisplayAttributes{
+			Name:  fields.FieldName,
+			Value: "default,field1,field2",
+		},
+		Default: []string{"default"},
+	}
+}
+
+// NewHandler instantiates a Handler to be embedded in your config.
+func NewHandler(fields *Fields) Handler {
+	return &handler{
+		fields: fields,
+	}
+}
+
+// Handler is an interface for the helper methods you get on your
+// config when you embed the Handler.
+type Handler interface {
+	GetAliasMetadata() []string
+	ParseAliasMetadata(data *framework.FieldData) error
+	PopulateDesiredAliasMetadata(auth *logical.Auth, fieldValues map[string]string) error
+}
+
+type handler struct {
+	// AliasMetadata is an explicit list of all the user's configured
+	// fields that are being added to alias metadata. It will never
+	// include the "default" parameter, and instead includes the actual
+	// fields behind "default", if selected. If it has never been set,
+	// the pointer will be nil.
+	AliasMetadata []string `json:"alias_metadata"`
+
+	// fields is a list of the configured default and available
+	// fields. It's intentionally not jsonified.
+	fields *Fields
+}
+
+// GetAliasMetadata is intended to be used on config reads.
+// It gets an explicit list of all the user's configured
+// fields that are being added to alias metadata. It will never
+// include the "default" parameter, and instead includes the actual
+// fields behind "default", if selected.
+func (h *handler) GetAliasMetadata() []string {
+	if h.AliasMetadata == nil {
+		return h.fields.Default
+	}
+	return h.AliasMetadata
+}
+
+// ParseAliasMetadata is intended to be used on config create/update.
+// It takes a user's selected fields (or lack thereof),
+// converts it to a list of explicit fields, and adds it to the handler
+// for later storage.
+func (h *handler) ParseAliasMetadata(data *framework.FieldData) error {
+	userProvided, ok := data.GetOk(h.fields.FieldName)
+	if !ok {
+		// Nothing further to do here.
+		return nil
+	}
+
+	// uniqueFields protects against weird edge cases like if
+	// a user provided "default,field1,field2,default".
+	uniqueFields := make(map[string]bool)
+	for _, field := range userProvided.([]string) {
+		if field == "default" {
+			// Add the fields that "default" represents, rather
+			// than the explicit field.
+			for _, dfltField := range h.fields.Default {
+				uniqueFields[dfltField] = true
+			}
```

Comment:
One thing to consider here is if we should be storing a nil slice if only the "default" case is passed. That will allow us to update the default set in future versions of Vault without having to worry about overriding storage. What do you think?

Reply:
Makes sense to me, and is effectively the current behavior with hard-coded metadata assignments.

Reply:
If we remove items, yes, but not if we are adding

Reply:
I like that approach as well.

Reply:
https://github.com/hashicorp/vault/pull/8783/commits/7c5cae53093a7aef21fcd5be9c4363446d6c546a